### PR TITLE
feat: Add build for Unity platform

### PR DESCRIPTION
**Pull request by cprice404**
This commit makes a small change to the csproj file that allows us
to specify DefineConstants=USE_GRPC_WEB for any target framework.
That allows us to do a netframework20 + grpc-web build, which is
necessary for compatibility with the Unity platform.

We also add a step to the release action to do a build with this
configuration, and attach it to the github release.

### CODE REVIEW

File: .github/workflows/on-push-to-release-branch.yaml [hunk 1]
```diff
@@ -57,3 +57,21 @@ jobs:
...
           popd
+          
+      - name: Build for Unity
+        run: |
+          set -x
+          pushd src/Momento.Sdk
+            rm -rf bin
+            VERSION="${{ needs.release.outputs.version }}"
+            echo "version: ${VERSION}"
+            dotnet build --configuration Release -f netframework20 -p:DefineConstants=USE_GRPC_WEB
```
Comment (malandis) [inline .github/workflows/on-push-to-release-branch.yaml:68]: netframework20 or netstandard20?

File: .github/workflows/on-push-to-release-branch.yaml [hunk 2]
```diff
@@ -57,3 +57,21 @@ jobs:
...
+            DLL_FILE=./bin/Release/Momento.Sdk.${VERSION}.dll
+            ARCHIVE_FILE_NAME=Momento.Sdk.Unity.${VERSION}.dll
+            AUTH="Authorization: token ${{ secrets.MOMENTO_MACHINE_USER_GITHUB_TOKEN }}"
+            LATEST_RELEASE=$(curl -sH "$AUTH" https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/v${VERSION})
+            RELEASE_ID=$(echo $LATEST_RELEASE | jq -r .id)
+            GH_ASSET="https://uploads.github.com/repos/${GITHUB_REPOSITORY}/releases/${RELEASE_ID}/assets?name=${ARCHIVE_FILE_NAME}"
+            echo $GH_ASSET
+            curl --data-binary @$DLL_FILE -H "$AUTH" -H "Content-Type: application/octet-stream" $GH_ASSET
```
Comment (malandis) [inline .github/workflows/on-push-to-release-branch.yaml:76]: We don't need to include the other dependencies?

**Review by malandis — COMMENTED** [review]
Couple ?s

File: .github/workflows/on-push-to-release-branch.yaml (hunk 1 above)
Comment (cprice404) [reply .github/workflows/on-push-to-release-branch.yaml:68]: derp

Comment (cprice404) [inline .github/workflows/on-push-to-release-branch.yaml:68]:
```suggestion
            dotnet build --configuration Release -f netstandard20 -p:DefineConstants=USE_GRPC_WEB
```

File: .github/workflows/on-push-to-release-branch.yaml (hunk 2 above)
Comment (cprice404) [reply .github/workflows/on-push-to-release-branch.yaml:76]: nope, we don't

**Review by malandis — APPROVED** [review]

**Pull request merged**